src/wp-includes/html-api/class-wp-html-doctype-info.php

```diff
@@ -0,0 +1,269 @@
+<?php
+/**
+ * HTML API: WP_HTML_Doctype_Info class
+ *
+ * @package WordPress
+ * @subpackage HTML-API
+ * @since 6.7.0
+ */
+
+/**
+ * Core class used by the HTML API processor to represent a DOCTYPE declaration.
+ *
+ * @since 6.7.0
+ */
+class WP_HTML_Doctype_Info {
+	/**
+	 * The name of the DOCTYPE.
+	 *
+	 * @var string|null
+	 */
+	private $name;
+
+	/**
+	 * The public identifier of the DOCTYPE.
+	 *
+	 * @var string|null
+	 */
+	private $public_identifier;
+
+	/**
+	 * The system identifier of the DOCTYPE.
+	 *
+	 * @var string|null
+	 */
+	private $system_identifier;
+
+	/**
+	 * Whether the DOCTYPE token force-quirks flag is set.
+	 *
+	 * @var bool
+	 */
+	private $force_quirks_flag;
```

Comment:
perhaps `$forces_quirks_flag` might align better with the nuance? _this declaration forces a document into quirks mode_

Reply:
I'd prefer to keep this `$force_quirks_flag` which aligns with the spec.

Reply:
I've re-read some of your other related comments and I think I agree after all. I'm going to re-work some of this to establish the resulting quirks mode on construction instead of re-running the algorithm in the `get_quirks_mode` function.